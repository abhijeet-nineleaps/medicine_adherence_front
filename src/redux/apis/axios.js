import defaultAxios from 'axios';
import { API_URL } from '../../repositories/var';

const axios = defaultAxios.create({
  baseURL: `${API_URL}`,
  headers: {'Content-Type': 'application/json'},
});

export default axios;
