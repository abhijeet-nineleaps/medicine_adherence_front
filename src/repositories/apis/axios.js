import defaultAxios from 'axios';
import { API_URL } from '../var';

const axios = defaultAxios.create({
  baseURL: `${API_URL}`,
  headers: {'Content-Type': 'application/json'},
});

export default axios;
