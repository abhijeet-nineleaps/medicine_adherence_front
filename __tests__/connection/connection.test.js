import React from 'react';
import renderer from 'react-test-renderer';
import enableHooks from 'jest-react-hooks-shallow';
import Enzyme, {shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
Enzyme.configure({adapter: new Adapter()});
import checkConnectivity from '../../src/connection/checkConnectivity';
import toJson from 'enzyme-to-json';
import axios from '../../src/repositories/apis/axios';
enableHooks(jest);

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useFocusEffect: jest.fn(),
  useEffect: jest.fn(),
}));
jest.mock('@react-native-google-signin/google-signin/lib/commonjs', () => ({
  default: jest.fn(),
}));
jest.mock('@react-native-community/netinfo', () => ({
  NetInfo: {
    addEventListener: () => ({}),
  },
}));
describe('check connectivity', () => {
  it('test', () => {
    const wrapper = shallow(<checkConnectivity />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('test the only function', () => {
    const wrapper = renderer.create(<checkConnectivity />);
    const inst = wrapper.getInstance();
    expect(inst?.addEventListener()).toMatchSnapshot();
  });
  it('test getusermeds', async () => {
    const payload = 'payload';
    jest
      .spyOn(axios, 'get')
      .mockImplementation(jest.fn(() => Promise.resolve({data: 'dfghjk'})));
    checkConnectivity(payload);
  });
});
