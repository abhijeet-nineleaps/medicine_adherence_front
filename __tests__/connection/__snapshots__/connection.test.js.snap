// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check connectivity test 1`] = `<checkConnectivity />`;

exports[`check connectivity test the only function 1`] = `undefined`;
